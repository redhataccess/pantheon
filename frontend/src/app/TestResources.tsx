import { IAppState } from './app';

const fakeState: IAppState = {
    isNavOpen: true,
    userAuthenticated: true,
    username: "demo"
}

export { fakeState }