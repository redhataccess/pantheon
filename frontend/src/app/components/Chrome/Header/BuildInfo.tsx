import React, { Component } from 'react';
import { Link } from "react-router-dom";

class BuildInfo extends Component {
    public state = {
        buildDate: '',
        commitHash: '',
        commitText: ''
    };

    public componentDidMount() {
        this.getBuildInfo()
    }

    public render() {
        return (
            <React.Fragment>
                <div className="column-view">
                    {!this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && <Link to=""
                        onClick={this.commitRedirect}>
                        {this.state.buildDate}
                    </Link>}
                    {this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && this.state.buildDate}
                </div>
            </React.Fragment>
        );
    }

    public getBuildInfo() {
        const backend = "/pantheon/builddate.json"
        if (this.state.buildDate === '') {
            fetch(backend)
                .then(response => response.json())
                .then(responseJSON => this.setState({ buildDate: "Build Date: " + responseJSON.buildDate, commitHash: responseJSON.commitHash }))
        }
    };

    private commitRedirect = () => {
        window.location.href = "https://github.com/redhataccess/pantheon/commit/" + this.state.commitHash
    }
}

export { BuildInfo }