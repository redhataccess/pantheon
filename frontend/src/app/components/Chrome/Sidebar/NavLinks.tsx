import React, { Component } from 'react';
import { NavItem, NavExpandable, NavList } from '@patternfly/react-core';
import { Link } from "react-router-dom";
import fetch from 'isomorphic-fetch';

// BASE is used in the fetch call to check if isLoggedIn or isAdmin. It currently breaks the Navlinks.
// only search is displayed when BASE is consumed.
// const BASE = process.env.BROWSER? '': `http://localhost`;
class NavLinks extends Component<any, any> {

  constructor(props) {
    super(props)
    this.state = {
      activeGroup: '',
      activeItem: '',
      gotUserInfo: false,
      isAdmin: false,
      isDropdownOpen: false,
      isKebabDropdownOpen: false,
      isLoggedIn: false,
    }
  }

  public componentDidMount() {
    if (!this.state.isLoggedIn) {
      fetch("/system/sling/info.sessionInfo.json")
        .then(response => response.json())
        .then(responseJSON => {
          if (responseJSON.userID !== 'anonymous') {
            this.setState({ moduleText: 'New Module' })
            this.setState({ isLoggedIn: true })
          }
          if (responseJSON.userID === 'admin') {
            this.setState({ isAdmin: true })
          }
        })
    }
    this.setState({ activeGroup: 'grp-1', activeItem: 'grp-1_itm-1' });
  }
  public render() {
    const browserText = 'Content Browser'
    const consoleText = 'Web Console'
    const gitText = 'Git Import'
    const moduleText = 'New Module'
    const productText = 'New Product'
    const productsText = 'Product Listing'
    const searchText = 'Search'
    const slingHomeText = 'Sling Welcome'

    return (
      <React.Fragment>
        <NavList>
          <NavExpandable title="Modules" groupId="grp-1" isActive={this.state.activeGroup === 'grp-1'} isExpanded={true} data-testid="navLink_modules">
            <NavItem groupId="grp-1" itemId="grp-1_itm-1" isActive={this.state.activeItem === 'grp-1_itm-1'}>
              <Link to='/search' data-testid="navLink_search">{searchText}</Link>
            </NavItem>
            {(this.state.isLoggedIn) && (<NavItem groupId="grp-1" itemId="grp-1_itm-2" isActive={this.state.activeItem === 'grp-1_itm-2'}>
              <Link to='/module'>{moduleText}</Link>
            </NavItem>)}
            {(this.state.isLoggedIn) && (<NavItem groupId="grp-1" itemId="grp-1_itm-3" isActive={this.state.activeItem === 'grp-1_itm-3'}>
              <Link to='/git'>{gitText}</Link>
            </NavItem>)}
          </NavExpandable>
          {(this.state.isLoggedIn) && (<NavExpandable title="Products" groupId="grp-2" isActive={this.state.activeGroup === 'grp-2'}>
            <NavItem groupId="grp-2" itemId="grp-2_itm-1" isActive={this.state.activeItem === 'grp-2_itm-1'}>
              <Link to='/products'>{productsText}</Link>
            </NavItem>
            {(productText.length > 0) && (<NavItem groupId="grp-2" itemId="grp-2_itm-2" isActive={this.state.activeItem === 'grp-2_itm-2'}>
              <Link to='/product'>{productText}</Link>
            </NavItem>)}
          </NavExpandable>)}
          {(this.state.isLoggedIn) && (this.state.isAdmin) && (<NavExpandable title="Admin Panel" groupId="grp-3" isActive={this.state.activeGroup === 'grp-3'}>
            <NavItem groupId="grp-3" itemId="grp-3_itm-1" isActive={this.state.activeItem === 'grp-3_itm-1'} preventDefault={false} component='a'>
              <a href='/starter/index.html' target='_blank'>{slingHomeText}</a>
            </NavItem>
            <NavItem groupId="grp-3" itemId="grp-3_itm-2" isActive={this.state.activeItem === 'grp-3_itm-2'} preventDefault={false} component='a'>
              <a href='/bin/browser.html' target='_blank'>{browserText}</a>
            </NavItem>
            <NavItem groupId="grp-3" itemId="grp-3_itm-3" isActive={this.state.activeItem === 'grp-3_itm-3'} preventDefault={false} component='a'>
              <a href='/system/console/bundles.html' target='_blank'>{consoleText}</a>
            </NavItem>
          </NavExpandable>)}
        </NavList>
      </React.Fragment>
    );
  }
}

export { NavLinks }