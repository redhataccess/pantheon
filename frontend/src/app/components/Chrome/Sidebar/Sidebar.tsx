import * as React from 'react';
import { NavLink } from 'react-router-dom';
import {
  PageSidebar,
} from '@patternfly/react-core';
import { routes } from '../../../routes';

export interface ISideBarProps {
  isNavOpen: boolean;
}

export const Sidebar: React.FunctionComponent<ISideBarProps> = ({isNavOpen}) => {
  const outputNavLinks = () => {
    return routes.map(route => {
      return (
        <li key={route.path}>
          <NavLink to={route.path} className="pf-c-nav__link">{route.label}</NavLink>
        </li>
      );
    });
  }

  const nav = () => (
    <aside className="pf-c-page__sidebar">
      <nav className="pf-c-nav" id="page-layout-default-nav-primary-nav" aria-label="Primary Nav Default Example">
        <ul className="pf-c-nav__list">
          {outputNavLinks()}
        </ul>
      </nav>
    </aside>
  );

  return <PageSidebar nav={nav()} isNavOpen={isNavOpen} />;
}