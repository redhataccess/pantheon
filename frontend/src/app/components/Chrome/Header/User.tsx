import React, { Component } from 'react'
import { Link } from "react-router-dom"
import { IAppState } from '@app/app'

class User extends Component<IAppState, any> {
    public render() {
        return (
            <React.Fragment>
                <Link to={this.props.userAuthenticated ? '' : '/login'}
                    onClick={this.conditionalRedirect}>
                    {this.props.userAuthenticated ? 'Log Out [' + this.props.username + ']' : 'Log In'}
                </Link>
            </React.Fragment>
        );
    }

    private conditionalRedirect = () => {
        if (this.props.userAuthenticated) {
            fetch('/system/sling/logout')
                .then(response => window.location.href = "/pantheon")
        }
    }
}

export { User }