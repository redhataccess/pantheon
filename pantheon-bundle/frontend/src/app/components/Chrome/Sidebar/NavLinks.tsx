import React, { Component } from 'react'
import { NavItem, NavExpandable, NavList } from '@patternfly/react-core'
import { Link } from "react-router-dom"
import { IAppState } from '@app/app'

// BASE is used in the fetch call to check if isLoggedIn or isAdmin. It currently breaks the Navlinks.
// only search is displayed when BASE is consumed.
// const BASE = process.env.BROWSER? '': `http://localhost`;
class NavLinks extends Component<IAppState, any> {

  constructor(props) {
    super(props)
    this.state = {
      activeGroup: '',
      activeItem: '',
      expandableClickedGroup: '',
      expandableClickedItem: '',
      gotUserInfo: false,
      isDropdownOpen: false,
      isKebabDropdownOpen: false
    }
  }
  public render() {
    const browserText = 'Content Browser'
    const consoleText = 'Web Console'
    const gitText = 'Git Import'
    const productText = 'New Product'
    const productsText = 'Product Listing'
    const searchText = 'Search'
    const slingHomeText = 'Sling Welcome'

    return (
      <React.Fragment>
        <NavList onSelect={this.onExpandableSelect} id="nav-primary-expandable">
          <NavExpandable title="Modules" groupId="grp-1" isActive={this.state.activeGroup === 'grp-1'} isExpanded={true} data-testid="navLink_modules">
            <NavItem
              groupId="grp-1"
              itemId="grp-1_itm-1"
              isActive={this.state.activeGroup === 'grp-1' && this.state.activeItem === 'grp-1_itm-1'}
              onClick={this.handleItemOnclick}
            >
              <Link to='/search' data-testid="navLink_search">{searchText}</Link>
            </NavItem>
            {(this.props.userAuthenticated) &&
              (<NavItem
                groupId="grp-1"
                itemId="grp-1_itm-2"
                isActive={this.state.activeItem === 'grp-1_itm-2'}
                onClick={this.handleItemOnclick}
              >
                <Link to='/git'>{gitText}</Link>
              </NavItem>)}
          </NavExpandable>
          {(this.props.userAuthenticated) && (<NavExpandable title="Products" groupId="grp-2" isActive={this.state.activeGroup === 'grp-2'}>
            <NavItem
              groupId="grp-2"
              itemId="grp-2_itm-1"
              isActive={this.state.activeGroup === 'grp-2' && this.state.activeItem === 'grp-2_itm-1'}
              onClick={this.handleItemOnclick}
            >
              <Link to='/products'>{productsText}</Link>
            </NavItem>
            {(productText.length > 0) &&
              (<NavItem
                groupId="grp-2"
                itemId="grp-2_itm-2"
                isActive={this.state.activeGroup === 'grp-2' && this.state.activeItem === 'grp-2_itm-2'}
                onClick={this.handleItemOnclick}
              >
                <Link to='/product'>{productText}</Link>
              </NavItem>)}
          </NavExpandable>)}
          {(this.props.userAuthenticated) && this.props.isAdmin && (<NavExpandable title="Admin Panel" groupId="grp-3" isActive={this.state.activeGroup === 'grp-3'}>
            <NavItem
              groupId="grp-3"
              itemId="grp-3_itm-1"
              isActive={this.state.activeGroup === 'grp-3' && this.state.activeItem === 'grp-3_itm-1'}
              preventDefault={false}
              component='a'
              onClick={this.handleItemOnclick}
            >
              <a href='/starter/index.html' target='_blank'>{slingHomeText}</a>
            </NavItem>
            <NavItem
              groupId="grp-3"
              itemId="grp-3_itm-2"
              isActive={this.state.activeGroup === 'grp-3' && this.state.activeItem === 'grp-3_itm-2'}
              preventDefault={false}
              component='a'
              onClick={this.handleItemOnclick}
            >
              <a href='/bin/browser.html' target='_blank'>{browserText}</a>
            </NavItem>
            <NavItem
              groupId="grp-3"
              itemId="grp-3_itm-3"
              isActive={this.state.activeGroup === 'grp-3' && this.state.activeItem === 'grp-3_itm-3'}
              preventDefault={false}
              component='a'
              onClick={this.handleItemOnclick}
            >
              <a href='/system/console/bundles.html' target='_blank'>{consoleText}</a>
            </NavItem>
          </NavExpandable>)}
        </NavList>
      </React.Fragment>
    );
  }
  private onExpandableSelect = result => {
    this.setState({
      activeGroup: result.groupId,
      activeItem: result.itemId
    });
  };

  private handleItemOnclick = (
    e: React.FormEvent<HTMLInputElement>,
    itemId: number | string,
    groupId: number | string
  ): void => {
    this.setState({
      expandableClickedGroup: groupId,
      expandableClickedItem: itemId
    });
  };
}

export { NavLinks }