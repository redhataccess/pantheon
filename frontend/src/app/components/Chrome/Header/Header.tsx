import React from 'react'
import { PageHeader } from '@patternfly/react-core'
import { Brand } from '../Header/Brand/Brand'
import { User }  from '../Header/User'
import { IAppState } from '@app/app'

export interface IHeaderProps {
    isNavOpen: boolean
    onNavToggle: any
    appState: React.PropsWithChildren<IAppState>
}

export const Header: React.FunctionComponent<IHeaderProps> = ({
    isNavOpen,
    onNavToggle,
    appState
}) => {
    return (
        <PageHeader
            logo={<Brand />}
            toolbar={<User {...appState} />}
            showNavToggle={true}
            isNavOpen={isNavOpen}
            onNavToggle={onNavToggle}
        />
    )
}