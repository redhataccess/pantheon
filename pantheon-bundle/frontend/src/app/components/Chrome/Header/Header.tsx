import React from 'react'
import { PageHeader, Brand } from '@patternfly/react-core'
// import { Brand } from '../Header/Brand/Brand'
import { User }  from '../Header/User'
import { IAppState } from '@app/app'
import logo from '../../../images/Pantheon2-logo-white.png';

export interface IHeaderProps {
    isNavOpen: boolean
    onNavToggle: any
    appState: React.PropsWithChildren<IAppState>
}

export const Header: React.FunctionComponent<IHeaderProps> = ({
    isNavOpen,
    onNavToggle,
    appState
}) => {
    return (
        <PageHeader
            logo={<Brand src={logo} alt="Pantheon" className="PageHeader__Brand" />}
            logoProps={
                {
                'href': '/',
                // @todo Replace in CSS file when a project stylesheet has been setup
                'style': {'maxWidth': '220px'}
                }
            }
            toolbar={<User {...appState} />}
            showNavToggle={true}
            isNavOpen={isNavOpen}
            onNavToggle={onNavToggle}
        />
    )
}