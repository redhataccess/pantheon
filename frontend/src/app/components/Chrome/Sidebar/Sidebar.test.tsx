import React from 'react';
import { Sidebar }  from './Sidebar';
import { NavList } from '@patternfly/react-core';
import { NavLinks }  from './NavLinks';
import { BuildInfo }  from '../Header/BuildInfo';
import { mount, shallow } from 'enzyme';
import { HashRouter as Router } from 'react-router-dom';
import "isomorphic-fetch"
import { mockStateUser } from '@app/TestResources'

describe('Sidebar tests', () => {
  test('should render Sidebar component', () => {
    const view = shallow(<Sidebar isNavOpen={true} appState={mockStateUser} />);
    expect(view).toMatchSnapshot();
  });

  it('should render a NavList', () => {
    const wrapper = mount(<Router><Sidebar isNavOpen={true} appState={mockStateUser} /></Router>);
    const navList = wrapper.find(NavList);
    expect(navList.exists()).toBe(true)
  });

  it('should render a NavLinks component', () => {
    const wrapper = mount(<Router><Sidebar isNavOpen={true} appState={mockStateUser} /></Router>);
    const navLinks = wrapper.find(NavLinks);
    expect(navLinks.exists()).toBe(true)
  });

});
