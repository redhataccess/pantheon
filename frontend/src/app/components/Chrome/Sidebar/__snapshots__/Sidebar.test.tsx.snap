// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar tests should render Sidebar component 1`] = `
<PageSidebar
  isNavOpen={true}
  nav={
    <aside
      className="pf-c-page__sidebar"
    >
      <Unknown
        aria-label="Primary Nav Default"
        className="pf-c-nav"
        id="page-layout-default-nav-primary-nav"
      >
        <NavList
          ariaLeftScroll="Scroll left"
          ariaRightScroll="Scroll right"
          className="pf-c-nav__list"
          variant="default"
        >
          <NavLinks
            isAdmin={false}
            isNavOpen={true}
            userAuthenticated={true}
            username="demo"
          />
        </NavList>
      </Unknown>
    </aside>
  }
/>
`;
